tests/e2e/components_test.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	return &componentCtx, nil
+}
+
+func (c *ComponentTestCtx) ValidateComponentEnabled(t *testing.T) {
+	g := c.NewWithT(t)
+
+	g.Update(
+		gvk.DataScienceCluster,
+		c.DSCName,
+		testf.Transform(`.spec.components.%s.managementState = "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Managed),
+	).Eventually().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
+		jq.Match(`.spec.components.%s.managementState == "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Managed),
+	)
+
+	g.List(gvk.DataScienceCluster).Eventually().Should(And(
+		HaveLen(1),
+		HaveEach(And(
+			jq.Match(`.spec.components.%s.managementState == "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Managed),
+			jq.Match(`.status.conditions[] | select(.type == "%sReady") | .status == "%s"`, c.GVK.Kind, metav1.ConditionTrue),
+		)),
+	))
+
+	g.List(c.GVK).Eventually().Should(And(
+		HaveLen(1),
+		HaveEach(And(
+			jq.Match(`.metadata.ownerReferences[0].kind == "%s"`, gvk.DataScienceCluster.Kind),
+			jq.Match(`.status.conditions[] | select(.type == "Ready") | .status == "%s"`, metav1.ConditionTrue),
+		)),
+	))
+}
+
+func (c *ComponentTestCtx) ValidateOperandsOwnerReferences(t *testing.T) {
+	g := c.NewWithT(t)
+
+	g.List(
+		gvk.Deployment,
+		client.InNamespace(c.ApplicationNamespace),
+		client.MatchingLabels{labels.PlatformPartOf: strings.ToLower(c.GVK.Kind)},
+	).Eventually().Should(And(
+		Not(BeEmpty()),
+		HaveEach(
+			jq.Match(`.metadata.ownerReferences[0].kind == "%s"`, c.GVK.Kind),
+		),
+	))
+}
+
+func (c *ComponentTestCtx) ValidateUpdateDeploymentsResources(t *testing.T) {
+	g := c.NewWithT(t)
+
+	deployments := g.List(
+		gvk.Deployment,
+		client.InNamespace(c.ApplicationNamespace),
+		client.MatchingLabels{
+			labels.PlatformPartOf: strings.ToLower(c.GVK.Kind),
+		},
+	).Eventually().ShouldNot(
+		BeEmpty(),
+	)
+
+	for _, d := range deployments {
+		t.Run("deployment_"+d.GetName(), func(t *testing.T) {
+			replicas, err := jq.ExtractValue[int](d, `.spec.replicas`)
+			g.Expect(err).ShouldNot(HaveOccurred())
+
+			expectedReplica := replicas + 1
+			if replicas > 1 {
+				expectedReplica = 1
+			}
+
+			g.Update(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+				testf.Transform(`.spec.replicas = %d`, expectedReplica),
+			).Eventually().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+
+			g.Get(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+			).Eventually().Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+
+			g.Get(
+				gvk.Deployment,
+				client.ObjectKeyFromObject(&d),
+			).Consistently().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
+				jq.Match(`.spec.replicas == %d`, expectedReplica),
+			)
+		})
+	}
+}
+
+func (c *ComponentTestCtx) ValidateComponentDisabled(t *testing.T) {
+	g := c.NewWithT(t)
+
+	g.List(c.GVK).Eventually().ShouldNot(
+		BeEmpty(),
+	)
+
+	g.Update(
```

Comment:
Thoughts. From testing point of view updating and then checking updated result is totally fine. But in these cases it leads to duplication of the expressions. Would it be sufficient if Update returns EventuallyErr, like Delete and the assertments of consequences of the update would catch the problems? May be to check if Update itself works as expected separately is enough?